gke-disk-image-builder/script/startup.sh

```diff
@@ -84,14 +84,33 @@ function process_snapshots() {
   for snapshot in "${snapshots[@]}"; do
     echo Processing $snapshot
 
-    sudo ctr -n k8s.io snapshot view tmp_$snapshot $snapshot
-    # Check if the view was successfully created
-    if [ $? -ne 0 ]; then
-      echo Failed to create snapshot view for $snapshot. Please rerun the tool to try it again.
+    # Add retry in case `ctr snapshot view` fails silently
+    local retries=5
+    while [ ${retries} -ge 0 ]; do 
+      ((retries--))
+      sudo ctr -n k8s.io snapshot view tmp_$snapshot $snapshot
+      # Check if the view was successfully created
+      if [ $? -ne 0 ]; then
+        echo "Failed to create snapshot view for $snapshot. Will retry. ${retries} retries left."
+        continue
+      fi
+      
+      original_path=$(sudo ctr -n k8s.io snapshot mount /tmp_$snapshot tmp_$snapshot | grep -oP '/\S+/snapshots/[0-9]+/fs' | tr ':' '\n' | head -n 1)
+      if [[ -n "$original_path" ]]; then
+        break
+      fi
+      echo "Failed to get mount point for tmp_$snapshot. Will retry. ${retries} retries left."
+      echo "All snapshots:"
+      sudo ctr -n k8s.io snapshot list
+      sudo ctr -n k8s.io snapshot rm tmp_$snapshot
+      sleep 1
+    done
+
+    if [ ${retries} -lt 0 ]; then
+      echo Failed to get snapshot directory for snapshot $snapshot. Please rerun the tool to try it again.
```

Comment:
With `((retries--))` at the top of the loop, the last iteration will leave the `retries` as `-1`, regardless of whether or not it succeeded.

```
$ cat retries.sh 
#!/bin/bash

retries=5
while [ ${retries} -ge 0 ]; do
  ((retries--))
  if [ ${retries} -ge 0 ]; then
    echo "Trying again, ${retries} left"
    continue
  fi
  echo "Last retry, success"
done

if [ ${retries} -lt 0 ]; then
  echo "Oops, all retries exhausted"
fi 
$ ./retries.sh 
Trying again, 4 left
Trying again, 3 left
Trying again, 2 left
Trying again, 1 left
Trying again, 0 left
Last retry, success
Oops, all retries exhausted
```

Reply:
uh nice catch. I changed the entry condition of the while loop to >= 1. And changed the if-condition at the end to check `original_path`